src/libs/nav2_util/CMakeLists.txt

```diff
@@ -17,18 +17,43 @@ find_package(tf2 REQUIRED)
 
 include_directories(include)
 
-add_library(nav2_util STATIC
+add_library(costmap_lib SHARED
   src/costmap.cpp
 )
 
-ament_target_dependencies(nav2_util
+ament_target_dependencies(costmap_lib
```

Comment:
I don't think that costmap_lib depends on all of these. Please file an issue to look at the message types used by the costmap code and pare this list down to only the ones that it needs.

Reply:
Will address this. Thanks.